pkg/kubectl/cmd/util/openapi/openapi.go

```diff
@@ -17,236 +17,108 @@ limitations under the License.
 package openapi
 
 import (
-	"fmt"
-	"sort"
-	"strings"
+	openapi_v2 "github.com/googleapis/gnostic/OpenAPIv2"
 
 	"k8s.io/apimachinery/pkg/runtime/schema"
-)
-
-// Defines openapi types.
-const (
-	Integer = "integer"
-	Number  = "number"
-	String  = "string"
-	Boolean = "boolean"
-
-	// These types are private as they should never leak, and are
-	// represented by actual structs.
-	array  = "array"
-	object = "object"
+	"k8s.io/kube-openapi/pkg/util/proto"
 )
 
 // Resources interface describe a resources provider, that can give you
 // resource based on group-version-kind.
 type Resources interface {
-	LookupResource(gvk schema.GroupVersionKind) Schema
-}
-
-// SchemaVisitor is an interface that you need to implement if you want
-// to "visit" an openapi schema. A dispatch on the Schema type will call
-// the appropriate function based on its actual type:
-// - Array is a list of one and only one given subtype
-// - Map is a map of string to one and only one given subtype
-// - Primitive can be string, integer, number and boolean.
-// - Kind is an object with specific fields mapping to specific types.
-// - Reference is a link to another definition.
-type SchemaVisitor interface {
-	VisitArray(*Array)
-	VisitMap(*Map)
-	VisitPrimitive(*Primitive)
-	VisitKind(*Kind)
-	VisitReference(Reference)
+	LookupResource(gvk schema.GroupVersionKind) proto.Schema
 }
 
-// Schema is the base definition of an openapi type.
-type Schema interface {
-	// Giving a visitor here will let you visit the actual type.
-	Accept(SchemaVisitor)
-
-	// Pretty print the name of the type.
-	GetName() string
-	// Describes how to access this field.
-	GetPath() *Path
-	// Describes the field.
-	GetDescription() string
-	// Returns type extensions.
-	GetExtensions() map[string]interface{}
-}
+// groupVersionKindExtensionKey is the key used to lookup the
+// GroupVersionKind value for an object definition from the
+// definition's "extensions" map.
+const groupVersionKindExtensionKey = "x-kubernetes-group-version-kind"
 
-// Path helps us keep track of type paths
-type Path struct {
-	parent *Path
-	key    string
+// Definitions is an implementation of `Resources`. It looks for
```

Comment:
Do you want to say `document is an implementation of Resources.`?

Reply:
yes, thank you:-)